pkg/middlewares/aws_sqs_queue_policy_expander.go

```diff
@@ -0,0 +1,78 @@
+package middlewares
+
+import (
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+	"github.com/sirupsen/logrus"
+)
+
+// Explodes policy found in aws_sqs_queue.policy from state resources to dedicated resources
+type AwsSqsQueuePolicyExpander struct{}
+
+func NewAwsSqsQueuePolicyExpander() AwsSqsQueuePolicyExpander {
+	return AwsSqsQueuePolicyExpander{}
+}
+
+func (m AwsSqsQueuePolicyExpander) Execute(_, resourcesFromState *[]resource.Resource) error {
+	newList := make([]resource.Resource, 0)
+	for _, res := range *resourcesFromState {
+		// Ignore all resources other than sqs_queue
+		if res.TerraformType() != aws.AwsSqsQueueResourceType {
+			newList = append(newList, res)
+			continue
+		}
+
+		queue, _ := res.(*aws.AwsSqsQueue)
+		newList = append(newList, res)
+
+		if m.hasPolicyAttached(queue, resourcesFromState) {
+			queue.Policy = nil
+			continue
+		}
+
+		err := m.handlePolicy(queue, &newList)
+		if err != nil {
+			return err
+		}
+	}
+	*resourcesFromState = newList
+	return nil
+}
+
+func (m *AwsSqsQueuePolicyExpander) handlePolicy(queue *aws.AwsSqsQueue, results *[]resource.Resource) error {
+	if queue.Policy == nil || *queue.Policy == "" {
```

Comment:
This can be moved in `Execute()` before `if m.hasPolicyAttached(queue, resourcesFromState) {`. It will avoid useless slice traversal operations.